packages/performance/src/resources/trace.ts

```diff
@@ -145,26 +146,39 @@ export class Trace implements PerformanceTrace {
 
   /**
    * Increments a custom metric by a certain number or 1 if number not specified. Will create a new
-   * custom metric if one with the given name does not exist.
+   * custom metric if one with the given name does not exist. The value will be floored down to an
+   * integer.
    * @param counter Name of the custom metric
-   * @param num Increment by value
+   * @param numAsInteger Increment by value
    */
-  incrementMetric(counter: string, num = 1): void {
+  incrementMetric(counter: string, numAsInteger = 1): void {
     if (this.counters[counter] === undefined) {
       this.putMetric(counter, 0);
     }
-    this.counters[counter] += num;
+    const valueAsInteger: number = Math.floor(numAsInteger);
+    if (valueAsInteger < numAsInteger) {
+      consoleLogger.info(
+        `Metric value should be an Integer, setting the value as : ${valueAsInteger}.`
+      );
+    }
+    this.counters[counter] += valueAsInteger;
```

Comment:
Can the method be like the following?

```
incrementMetric(counter: string, numAsInteger = 1): void {
    if (this.counters[counter] === undefined) {
      this.putMetric(counter, numAsInteger);
    } else {
      this.putMetric(counter, this.counters[counter] + numAsInteger);
    }
}
```

Reply:
For the sake of readability, I would prefer to stick with the older format. Any key benefits with this optimization?

Reply:
Main benefit:

**Consistency and Accuracy**: We don't need to remember updating two APIs if the logic here has changed. Plus that consoleLogger message, validation logic, etc. won't be duplicated.

**Size**: Consolidated logic produces smaller SDK size.

**Readability**: A simplified version has better readability. (No need to worry about actual implementation of putMetric())

Reply:
Thanks James for clarifying, makes sense. Done.